### PR TITLE
add obb-handler

### PR DESCRIPTION
**Pull request by Tunkali**
Add obb filehandler to move sample files to public download folder.

### CODE REVIEW

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.IO;
+
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:5]: File is missing namespace. See the other files for reference

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        string destinationPath = "/sdcard/Download/DomeViewer";
+
+        try
+            {
+                System.IO.Directory.CreateDirectory(GetStoragePath());
+            }
+        catch (System.Exception e)
+            {
+                Debug.LogError(e.Message);
+            }
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:31]: Broken formatting

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        
+        try
+        {
+            // checkk obb folder existence
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:35]: Typo in comment ("checkk")

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            string path = "/sdcard/Download/DomeViewer";
+
+
+#if UNITY_ANDROID && !UNITY_EDITOR
+        // AndroidJavaClass Environment = new AndroidJavaClass("android.os.Environment");
+        // AndroidJavaObject directory = Environment.CallStatic<AndroidJavaObject>("getExternalStoragePublicDirectory", "Pictures");
+        // path = directory.Call<string>("getAbsolutePath");
+        // path = System.IO.Path.Combine(path, "Oculus");
+        // path = System.IO.Path.Combine(path, "Gallery");
+#else
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:77]: Why is this all commented out? Why would we ship that?

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            Debug.LogError("Error moving OBB-Files: " + e.Message);
+        }
+        #else
+        Debug.Log("Move obb files skipped");
+        #endif
+    }
+    private string GetStoragePath()
+        {
+            string path = "/sdcard/Download/DomeViewer";
+
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:69]: Formatting / indenting is broken

**Review by hybridherbst — CHANGES_REQUESTED** [review]
There are a number of formatting issues and likely crashes on specific devices introduced by this code. We will likely use another method to copy files, but for learning purposes please fix the issues outlined in this review.

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs (hunk 1 above)
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:2]: namespace is added and unneeded references removed

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    Mediafiles need to be in this location on Android (via sideload or via deployment with Meta Quest Developer Hub):
+    sdcard/Android/oob/com.example.myapp/ as *.jpg / *.png / *.mp4
+    #################################################################################################################
+*/
+
+    void Start()
+    {
+        #if UNITY_ANDROID && !UNITY_EDITOR
+        // Sourcepath OBB-Folder
+        string obbFolderPath = "/sdcard/Android/obb/"+Application.identifier;
```
[the remark this answers is not in the document]
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs]: The hardcoded path is focusing for using on quest. I am not aware about "softer" declarations. For devices without sdcard or different path the error will be catched and the file access won't be applied. I added a comment on this.

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs (hunk 2 above)
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs]: formatting is fixed

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs (hunk 3 above)
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs]: fixed

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                foreach (string obbFile in obbFiles)
+                {
+                    // get filename
+                    string fileName = Path.GetFileName(obbFile);
+
+                    // move current file to target location
+                    File.Move(obbFile, Path.Combine(destinationPath, fileName));
+                }
+
+                Debug.Log("All files are moved successfully.");
```
[the remark this answers is not in the document]
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs]: removed debug logs. thanks for the tip.

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs (hunk 4 above)
Comment (Tunkali) [reply projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs]: Examples for different path uses. i removed that.

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                }
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError("Error moving OBB-Files: " + e.Message);
+            }
+#else
+#endif
+        }
+        private string GetStoragePath()
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:55]: Missing newline, please format similar to other files

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            private string destinationPath = "/sdcard/Download/DomeViewer";
+
+            try
+                {
+                    System.IO.Directory.CreateDirectory(GetStoragePath());
+                }
+            catch (System.Exception e)
+                {
+                    Debug.LogError(e.Message);
+                }
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:28]: Incorrect indenting

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    public class OBBHandler : MonoBehaviour
+    {
+
+        // Mediafiles need to be in this location on Android (via sideload or via deployment with Meta Quest Developer Hub):
+        // sdcard/Android/oob/com.example.myapp/ as *.jpg / *.png / *.mp4
+
+        void Start()
+        {
+#if UNITY_ANDROID && !UNITY_EDITOR
+
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:15]: Extra line, please remove. Stick to how other files are formatted

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 11]
```diff
@@ -0,0 +1,66 @@
+using UnityEngine;
+
+namespace pfc.Fulldome
+{
+
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:5]: Extra line, please remove

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    public class OBBHandler : MonoBehaviour
+    {
+
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:8]: Extra line, please remove

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        // Mediafiles need to be in this location on Android (via sideload or via deployment with Meta Quest Developer Hub):
+        // sdcard/Android/oob/com.example.myapp/ as *.jpg / *.png / *.mp4
+
+        void Start()
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:12]: Missing access modifier

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs (hunk 8 above)
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:55]: Can be static

**Review by hybridherbst — CHANGES_REQUESTED** [review]
More formatting and access modifier changes are needed. Thanks!

**Comment by hybridherbst** [comment]
Thanks.
PR is now superseded by https://github.com/prefrontalcortex/DomeTools/commit/647ed8732f92a1b3f290f4e1080f3128cd2d2549, the originally discussed approach of keeping files in StreamingAssets and playing them directly from there.

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+#else
+#endif
+        }
+
+        private string GetStoragePath()
+        {
+            string path = "/sdcard/Download/DomeViewer";
+
+#if UNITY_ANDROID && !UNITY_EDITOR
+#else
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:58]: This #else is not needed

File: projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+                else
+                {
+                    Debug.LogError("OBB-Folder not found");
+                }
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError("Error moving OBB-Files: " + e.Message);
+            }
+#else
```
Comment (hybridherbst) [inline projects/DomeViewer/Assets/Modules/FileSystem/OBBHandler.cs:49]: This #else is not needed

**Review by hybridherbst — COMMENTED** [review]
Some cleanup issues

**Pull request closed without merging**